backend/src/main/java/com/carffeine/carffeine/station/domain/station/Stations.java

```diff
@@ -0,0 +1,59 @@
+package com.carffeine.carffeine.station.domain.station;
+
+import com.carffeine.carffeine.station.domain.charger.ChargerType;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Stations {
+
+    private static final int NONE_CAPACITY = 0;
+
+    private final List<Station> stations;
+
+    private Stations(List<Station> stations) {
+        this.stations = new ArrayList<>(stations);
+    }
+
+    public static Stations from(List<Station> stations) {
+        return new Stations(stations);
+    }
+
+    private static void filterByCompanyNames(List<Station> stations, List<String> companyNames) {
+        if (!companyNames.isEmpty()) {
+            stations.removeIf(station -> !companyNames.contains(station.getCompanyName()));
+        }
+    }
+
+    private static void filterByChargerTypes(List<Station> stations, List<ChargerType> chargerTypes) {
+        if (!chargerTypes.isEmpty()) {
+            stations.removeIf(station -> station
+                    .getChargers()
+                    .stream()
+                    .noneMatch(charger -> chargerTypes.contains(charger.getType()))
+            );
+        }
+    }
+
+    private static void filterByCapacities(List<Station> stations, List<BigDecimal> capacities) {
+        if (!capacities.isEmpty()) {
+            stations.removeIf(station -> station
+                    .getChargers()
+                    .stream()
+                    .noneMatch(charger -> capacities.stream()
+                            .anyMatch(capacity -> capacity.compareTo(charger.getCapacity()) == NONE_CAPACITY))
```

Comment:
```suggestion
                            .anyMatch(capacity -> isSame(capacity, charger.getCapacity()))
```
라고 private 메서드를 정의하면 깔끔할 것 같아요 그리고 compareTo 를 해서 0이면 같은 값이라는 뜻이니 상수로 정의 하지 않는 것도 괜찮아보여요

Reply:
예리하시군요